### PR TITLE
docs: Add @defer API reference documentation

### PR DESCRIPTION
**Pull request by jessicajaniuk**
This adds the initial API reference documentation for the new defer blocks.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/CONTRIBUTING.md#commit
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Documentation content changes

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/blocks/core/defer.md [hunk 1]
```diff
@@ -1 +1,57 @@
-Placeholder content
+A type of [block](api/core/defer) that can be used to defer load components, directives and pipes used inside a component template.
```
Comment (jelbourn) [inline aio/content/blocks/core/defer.md:1]: I realized we might want to say "defer load the JavaScript for components..." to be more clear _what's_ being deferred

**Review by jelbourn — APPROVED** [review]
LGTM

File: aio/content/blocks/core/defer.md (hunk 1 above)
Comment (jessicajaniuk) [reply aio/content/blocks/core/defer.md]: Updated!

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 61fb6b8b4a7b8ef24c28fa2743f15140f671c2c9.

**Pull request closed without merging**